### ISSUE TITLE
Quoted messages included in forwarded conversations causing "repeated" sections.

### ISSUE THREAD

**Issue by DGrothe-PhD** [thunderbird-conversations/thunderbird-conversations#1753]
My setting in Thunderbird is: reply above message, put citation below the message:
![grafik](https://user-images.githubusercontent.com/64636863/141490096-7c57c8e0-c864-4f4c-ba5e-0da66759261d.png)

I installed the addon and tested it on an existing conversation which I clicked forwarding (but did not send actually) only to copy the content into an empty MS Word document. Therein I noticed that the older messages and conversation threads repeated several times, with the following pattern: 
Message A
Message B
On...wrote... Message A
Message C
On...wrote... Message B
On...wrote... Message A
Message D
On...wrote... Message C
On...wrote... Message B
On...wrote... Message A
...
This would be O(n^2) of conversation length if copied and printed.

To explain by an example, I started a short 'conversation' between two mailboxes I own. To better identify the sequence of messages, I chose one calendar month for each message text, Januar, Februar, März (January, February, March) and added prefix `~~` to the repeated messages.

## Observation
Hier ist eine Konversation, die Sie interessieren könnte!
 
Me
15:47

Liebe X,

im Januar gibt es oft Schnee.

Grüß dich
 
Friend
15:48

Ja, und im Februar regnet es dann manchmal.
Mach's gut bis dahin

~~On 12.11.21 15:47, X <some e-mail> wrote:
~~> Liebe Y,
~~>
~~> im Januar gibt es oft Schnee.
~~>
~~> Grüß dich
~~>
~~>
 
Me
15:49

Und im März geht es auf Ostern zu.

~~On 12.11.21 15:48, some e-mail wrote:
~~> Ja, und im Februar regnet es dann manchmal.
~~> Mach's gut bis dahin
~~>
~~> On 12.11.21 15:47, X <some e-mail> wrote:
~~> > Liebe Y,
~~> >
~~> > im Januar gibt es oft Schnee.
~~> >
~~> > Grüß dich
~~> >
~~> >

## My expectation:
Hier ist eine Konversation, die Sie interessieren könnte!
 
Me
15:47

Liebe X,

im Januar gibt es oft Schnee.

Grüß dich
 
Friend
15:48

Ja, und im Februar regnet es dann manchmal.
Mach's gut bis dahin

Me
15:49

Und im März geht es auf Ostern zu.

**Comment by Standard8** [comment]
Please check you're on the latest Conversations version (4.0.5) - you didn't put it in as requested.

Printing the conversation should be almost exactly the same as printing exactly what is displayed in the message preview area. 

Therefore if you're getting duplicates, then those duplicates should also be displayed in the message preview window.

I think one of the earlier 4.0.x versions did have an issue with not de-duping correctly.

**Comment by DGrothe-PhD** [comment]
yes, it is version 4.0.5. For printing, I am afraid it is not working at all - when I click on "Konversation drucken", nothing ever happens.

**Comment by Standard8** [comment]
Now I re-read this, I think your issue is that the quoted messages are included in the forwarding.

I think that would be difficult to change.

We can't simply remove any quoted items from a forwarded message, since sometimes people selectively quote messages and that would be necessary to maintain context when reading through the forwarded conversation.

I also think that trying to detect blocks that are full quotes versus blocks that are selectively quoted would be difficult - different email clients handle quoting in slightly different ways, plus html vs plain text adds in a lot of complexity.

In short, I don't think there is much we can easily do here. If people choose to not tidy up the quoted message, then there's not much we can do here - we need to preserve the data in case it is useful, but removing the data would be hard.
